website/addons/wiki/views.py

```diff
@@ -211,7 +214,7 @@ def wiki_page_content(wname, wver=None, **kwargs):
 
 
 @must_be_valid_project  # injects project
-@must_have_permission('write')  # injects user, project
+@must_have_permission_or_public_wiki  # injects user, project
```

Comment:
When kwargs are injected by decorators and used in the view, those kwargs should be explicitly listed in the function's signature.